core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/MethodConfigResolver.scala

```diff
@@ -57,19 +58,22 @@ object MethodConfigResolver {
         case _ => throw new RawlsException(s"MethodConfiguration ${methodConfig.namespace}/${methodConfig.name} input ${fqn} value is unavailable")
       }
     }
+
     val agoraInputs = workflow.inputs
+
     val missingInputs = agoraInputs.filter { case (fqn, workflowInput) => (!methodConfig.inputs.contains(fqn) || isAttributeEmpty(fqn)) && !workflowInput.optional }.keys
-    val extraInputs = methodConfig.inputs.filter { case (name, expression) => !agoraInputs.contains(name) }.keys
-    if (missingInputs.nonEmpty || extraInputs.nonEmpty) {
-      val message =
-        if (missingInputs.nonEmpty)
-          if (extraInputs.nonEmpty)
-            "is missing definitions for these inputs: " + missingInputs.mkString(", ") + " and it has extraneous definitions for these inputs: " + extraInputs.mkString(", ")
-          else
-            "is missing definitions for these inputs: " + missingInputs.mkString(", ")
-        else
-          "has extraneous definitions for these inputs: " + extraInputs.mkString(", ")
-      throw new RawlsException(s"MethodConfiguration ${methodConfig.namespace}/${methodConfig.name} ${message}")
+    val extraInputs = methodConfig.inputs.keys.filterNot { agoraInputs.contains }
+    val invalidInputs = methodConfig.inputs.collect {
+      case (name, AttributeString(expr)) if InputExpression.build(expr).isFailure => s"$name -> $expr"
```

Comment:
Does the function `resolveInputsForEntities` right beneath this one require fiddling with?